compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -693,13 +690,13 @@ fn check_mir_is_available<'tcx, I: Inliner<'tcx>>(
             // won't cause cycles on this.
             if !inliner.tcx().is_mir_available(callee_def_id) {
                 debug!("item MIR unavailable");
-                return Err("implementation limitation");
+                return Err("implementation limitation -- MIR unavailable");
             }
         }
         // These have no own callable MIR.
         InstanceKind::Intrinsic(_) | InstanceKind::Virtual(..) => {
             debug!("instance without MIR (intrinsic / virtual)");
-            return Err("implementation limitation");
+            return Err("implementation limitation -- cannot inline intrinsic");
```

Comment:
Actually we can inline intrinsics, but we don't want to.
https://github.com/rust-lang/rust/blob/e16a049adbf94d610787430b6efdf31d896dc5b6/compiler/rustc_mir_transform/src/cross_crate_inline.rs#L37-L43